### PR TITLE
refactor(query): first check privilege in SystemEngine get_full_data

### PR DESCRIPTION
**Pull request by TCeason**
I hereby agree to the terms of the CLA available at: https://docs.databend.com/dev/policies/cla/

## Summary

Optimize:

in system table engine , we can only get database that current user can be visibility.

## Tests

- [ ] Unit Test
- [x] Logic Test
- [ ] Benchmark Test
- [ ] No Test - _Explain why_

## Type of change

- [ ] Bug Fix (non-breaking change which fixes an issue)
- [ ] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that could cause existing functionality not to work as expected)
- [ ] Documentation Update
- [ ] Refactoring
- [x] Performance Improvement
- [ ] Other (please describe):

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/16421)

### CODE REVIEW

File: src/query/storages/system/src/columns_table.rs [hunk 1]
```diff
@@ -298,12 +289,62 @@ pub(crate) async fn dump_tables(
...
+    } else {
+        let catalog_dbs = visibility_checker.get_visibility_database();
+        if let Some(catalog_dbs) = catalog_dbs {
+            for (catalog_name, dbs) in catalog_dbs {
+                if catalog_name == CATALOG_DEFAULT {
+                    let mut catalog_db_ids = vec![];
+                    for (db_name, db_id) in dbs {
+                        if let Some(db_name) = db_name {
+                            let db_id = catalog
+                                .get_database(&tenant, db_name)
```
Comment (TCeason) [inline src/query/storages/system/src/columns_table.rs:301]: Now I can call get_database to get the db that user can be accessed.

But maybe we can provide a new API mget_database(db_names: &[string]) to reduce the meta call? @drmingdrmer

Comment (TCeason) [reply src/query/storages/system/src/columns_table.rs:301]: I can provide db_names and I want a Vec<DatabaseInfo>

Comment (drmingdrmer) [reply src/query/storages/system/src/columns_table.rs:301]: Merge this:
- https://github.com/datafuselabs/databend/pull/16425

**Review by drmingdrmer — APPROVED** [review]
> Reviewed 10 of 17 files at r2, 3 of 5 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/datafuselabs/databend/16421)* status: 12 of 19 files reviewed, all discussions resolved

**Comment by TCeason** [comment]
Cc @wubx You can test show columns with this optimize.

**Review by flaneur2020 — APPROVED** [review]
LGTM :rocket:

**Review by b41sh — APPROVED** [review]

**Pull request merged**